Frontend/ui-driver/src/Screens/ReferralScreen/Controller.purs

```diff
@@ -274,6 +280,11 @@ eval (BottomNavBarAction (BottomNavBar.OnNavigate item)) state = do
       exit $ SubscriptionScreen state
     _ -> continue state
 
+eval (ReferralQrRendered id) state = 
+  continueWithCmd state [ do
+    _ <- launchAff_ do void $ generateQR (getPlayStoreLink (getMerchant FunctionCall)) id 200 0 -- link for qr generation (app link for customer)
```

Comment:
try to handle the response from the function and move this `launchAff_` also to the function call.